datasources/github/github.go

```diff
@@ -87,6 +87,14 @@ func (GitHub) Recognize(_ context.Context, filenames []string) (timeline.Recogni
 
 // FileImport conducts an import of the data using this data source.
 func (c *GitHub) FileImport(ctx context.Context, filenames []string, itemChan chan<- *timeline.Graph, _ timeline.ListingOptions) error {
+	// If we are testing, close the channel to signal the end of the import, so we can
+	// count the total items sent.
+	defer func() {
+		if b, ok := ctx.Value("TLZ_TEST").(bool); ok && b {
+			close(itemChan)
+		}
+	}()
```

Comment:
Hmm. Is it possible to have the tests close the channel when they're done? Maybe in [`T.Cleanup`](https://pkg.go.dev/testing#T.Cleanup)?

Reply:
Yeah tricky. That wouldn't work for this case, afaict. I tried a few variations without much success.

The problem I have with the channel writer not being responsible for closing it, is is that I'm not able to count the total number of items sent. The caller (reader/test) knows how many items should be written, but a bug in the writer could send less and the test would block trying to read more, or the writer could send more and the reader would stop the count after the desired number, missing the bug.

Reply:
Ooo. I see. This isn't so bad, but it would be nice if it didn't require modifying the data sources like this. Let me think on it. :thinking:

Reply:
@mholt I forgot an important detail that solves the issue in a nicer way without the context hack: you can read from closed channels.

Reply:
Ohh very nice! Yes, closed channels can be read which is kinda great :)